Sources/PayPal/Buttons/PaymentButton.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+enum PayPalButtonType {
+    case payPal
+    case payPalCredit
+}
+
+/// Handles functionality shared across payment buttons
+public class PaymentButton: UIButton {
+
+    // asset identfier path for image and color button assets
+    private let bundleIdentifier = Bundle(identifier: "com.paypal.ios-sdk.PayPal")
+
+    // MARK: - Internal Helper Functions
+
+    func getButtonColor(for buttonType: PayPalButtonType) -> UIColor? {
+        switch buttonType {
+        case .payPal:
+            return UIColor(named: "PayPalGold", in: bundleIdentifier, compatibleWith: nil)
+        case .payPalCredit:
+            return UIColor(named: "PayPalDarkBlue", in: bundleIdentifier, compatibleWith: nil)
+        }
+    }
+
+    func getButtonLogo(for buttonType: PayPalButtonType) -> UIImage? {
+        switch buttonType {
+        case .payPal:
+            return UIImage(named: "PayPalLogo", in: bundleIdentifier, compatibleWith: nil)
+        case .payPalCredit:
+            return UIImage(named: "PayPalCreditLogo", in: bundleIdentifier, compatibleWith: nil)
+        }
+    }
+
+    func setLogoImageSize() -> UIEdgeInsets {
+        let buttonSize = frame.size
+        return UIEdgeInsets(
+            top: buttonSize.height / 4,
```

Comment:
Why divided by 4 here? Maybe we can move this # into a variable for proper naming.

Reply:
for now we decided to divide by 4 to make the logo half the size of the button (as we tested that was what looks nicest to us)
but probably in the future if we get exact requirement from design/UX then we'll go back in here to update this :smile: